core/iwasm/interpreter/wasm_loader.c

```diff
@@ -5428,7 +5515,24 @@ wasm_loader_get_const_offset(WASMLoaderContext *ctx, uint8 type, void *value,
         else
             operand_offset += 2;
     }
+
     if ((uint8 *)c == ctx->const_buf + ctx->num_const * sizeof(Const)) {
+        /* New constant, append to the const buffer */
+        if ((type == VALUE_TYPE_F64) || (type == VALUE_TYPE_I64)) {
+            bytes_to_increase = 2;
+        }
+        else {
+            bytes_to_increase = 1;
+        }
+
+        /* The max cell num of const buffer is 32768 since the valid index range
+         * is -32768 ~ -1. Return an invalid index 0 to indicate the buffer is
+         * full */
+        if (ctx->const_cell_num + bytes_to_increase > 32768) {
```

Comment:
Had better use `ctx->const_cell_num >= INT32_MAX - bytes_to_increase` to check integer overflow, as recommended by SDL guideline

Reply:
Done, thanks